src/core/server/metrics/integration_tests/metrics_service.test.ts

```diff
@@ -0,0 +1,161 @@
+        'collectionInterval',
+        otherTestSettings.ops.interval
+      );
+      expect(coreSetup.metrics).toHaveProperty('getOpsMetrics$');
+    });
+  });
+
+  describe('ops metrics logging configuration', () => {
+    it('does not log with logging set to quiet', async () => {
+      root = kbnTestServer.createRoot({ logging: { quiet: true } });
+      coreSetup = await root.setup();
+
+      expect(mockConsoleLog).not.toHaveBeenCalled();
+    });
+
+    it('logs at the correct level and with the correct context', async () => {
+      const testSettings = {
+        ops: {
+          interval: 500,
+        },
+        logging: {
+          silent: true, // set "true" in kbnTestServer
+          appenders: {
+            'custom-console': {
+              kind: 'console',
+              layout: {
+                highlight: false,
+                kind: 'pattern',
+                pattern: '%level|%logger',
+              },
+            },
+          },
+          root: {
+            appenders: ['custom-console', 'default'],
+            level: 'warn',
+          },
+          loggers: [
+            {
+              context: 'metrics.ops',
+              appenders: ['custom-console'],
+              level: 'debug',
+            },
+          ],
+        },
+        plugins: {
+          initialize: false,
+        },
+      };
+      root = kbnTestServer.createRoot({ ...testSettings });
+      coreSetup = await root.setup();
+
+      expect(mockConsoleLog).toHaveBeenCalledTimes(1);
+      const [level, logger] = mockConsoleLog.mock.calls[0][0].split('|');
+      expect(level).toBe('DEBUG');
+      expect(logger).toBe('metrics.ops');
+    });
+  });
+  describe('ops metrics logging content', () => {
+    it('logs memory, uptime, load and delay ops metrics in the message', async () => {
+      root = kbnTestServer.createRoot({ ...otherTestSettings });
+      coreSetup = await root.setup();
+
+      coreSetup.metrics.getOpsMetrics$().subscribe((opsMetrics) => {
+        testData = opsMetrics;
+      });
+      const expectedArray = ['memory:', 'uptime:', 'load:', 'delay:'];
+
+      expect(testData).toBeTruthy();
+      expect(mockConsoleLog).toHaveBeenCalledTimes(1);
+      const [message] = mockConsoleLog.mock.calls[0][0].split('|');
+
+      const messageParts = message.split(' ');
+      const testParts = [messageParts[0], messageParts[2], messageParts[4], messageParts[6]];
+      // the contents of the message are variable based on the process environment,
+      // so we are only performing assertions against parts of the string
+      expect(testParts).toEqual(expect.arrayContaining(expectedArray));
+    });
+
+    it('logs structured data in the log meta', async () => {
+      root = kbnTestServer.createRoot({ ...otherTestSettings });
+      coreSetup = await root.setup();
+
+      const [, meta] = mockConsoleLog.mock.calls[0][0].split('|');
+      expect(Object.keys(JSON.parse(meta).host.os.load)).toEqual(['1m', '5m', '15m']);
```

Comment:
I'm not sure it's worth adding all these tests, because `server_collector.test.ts` already covers metric creation and logging format tested by `get_ops_metrics_log.test.ts`.

Reply:
@restrry I added these because there weren't any jest integration tests that covered the `metric_service` itself. If you feel that they don't add value, I'll remove them.